### DISCUSSION TITLE
Cluster Controller Project

### DISCUSSION THREAD

**Discussion by jsiwek** [zeek/zeek#582] [Projects]
* Current [design doc](https://docs.google.com/document/d/1r0wXnihx4yESOpLJ87Wh2g1V-aHOFUkbgiFe8RHJpZo)
* [Roadmap](https://docs.google.com/spreadsheets/d/1oNYqHMU1b6FEfST8F86bdpv9Bpl6M58qq4AZULCIYlU) (out of date for >= 4.0)
* [Examples](https://docs.zeek.org/en/current/frameworks/supervisor.html) (Available as of Zeek 3.1.0)

Previous design/context (some parts may be outdated/historical by now):
* [Blog](https://blog.zeek.org/2019/03/beyond-brocontrol-new-process.html)
* [Design Doc](https://docs.google.com/document/u/1/d/e/2PACX-1vTA-kLzO_kEzaq4aHY6vCOjoCGWHaTWGKapfea64ga8RZ7qoElDUuaGvy6NnPe2UdNHq9mOdHKFRE9x/pub)

Notes:
* With the new approach, we can ship with a recommended systemd config, and include sandbox per this comment: https://github.com/zeek/zeek/pull/754#issuecomment-580652275

**Comment by jsiwek** [comment]
The initial framework for this will be in 3.1, but there's still more planned for 3.2.  Namely, the ability to run a supervised cluster over offline pcap file.

**Comment by rsmmr** [comment]
We can also start thinking about the evolution user interface for this: configuring clusters more automatically, and/or from config files, and/or expanding the `-j` option to allow specifying which/how  many processes to spawn.

**Comment by rsmmr** [comment]
The plan here is to develop this further over the next Zeek versions until we can eventually deprecate ZeekControl, if possible with 4.0. I added a roadmap doc to the ticket description, looking for feedback.

**Comment by klehigh** [comment]
A feature request for "Crash reports" would be to reduce the volume of individual email when a cluster crashes.  If we suffer a significant cluster failure, it generates a huge volume of emails (one per worker).  They are often not useful, as there is no core file.  Maybe have an option to not send an email if there is no core dump?

**Comment by rsmmr** [comment]
Looks like we're done for 3.2 here (with the reduced scope), moving to 4.0.
